### ISSUE TITLE
Inline functions with parameters in markup section confuse highlighter

### ISSUE THREAD

**Issue by Suges** [laosb/SvelteNova#7]
Hi there,

This plugin doesn't seem to be able to handle inline functions that have parameters, such as:
```svelte
<Checkbox bind:checked={checked} on:click={() => handleCheckBrand(id)} label={text} />
```
...in Nova, here's what happens:

<img width="621" alt="Screen Shot 2021-05-03 at 10 49 10 AM" src="https://user-images.githubusercontent.com/7140406/116892004-5a993e00-abfd-11eb-9552-b662ad7325c8.png">

So as you can see, once it encounters the function, syntax highlighting fails for the rest of the line (this doesn't happen in VS Code or SublimeText, for example).

But it _does_ work fine if the function has no parameter:
```svelte
<Checkbox bind:checked={checked} on:click={handleCheckBrand} label={text} />
```

<img width="553" alt="Screen Shot 2021-05-03 at 10 53 04 AM" src="https://user-images.githubusercontent.com/7140406/116892357-c24f8900-abfd-11eb-8fc5-f7da316531f0.png">

Thanks!

**Comment by laosb** [comment]
Thanks for reporting this. Looks like problem in syntax definitions, which is not something I'm good at. I may take a look when I'm free, but it's kind of beyond my capabilities.

**Comment by Suges** [comment]
I hear you! Appreciate the response.

**Comment by cuzzins** [comment]
Ran into this issue today so am assuming it hasn't been looked into yet? Is there any documentation related to the syntax format being used in the Syntax.xml file?

**Comment by Bluebie** [comment]
It gets really nasty when template literals are involved, the glitched highlighting continues on down the page affecting highlighting for the rest of the document after this tag.

<img width="788" alt="Screen Shot 2022-03-31 at 1 58 55 am" src="https://user-images.githubusercontent.com/12768/160866491-0eaf13f6-2a12-442e-8e48-6bb84525c98a.png">

Also, anyone know who I should complain to about the URLSearchParams definition being wrong? It does accept object's like that.

**Comment by laosb** [comment]
> Ran into this issue today so am assuming it hasn't been looked into yet? Is there any documentation related to the syntax format being used in the Syntax.xml file?

Panic has official documentation on this: https://docs.nova.app/syntax-reference/

If anyone is interested in improving that I would be more than happy to merge!

**Comment by laosb** [comment]
> Also, anyone know who I should complain to about the URLSearchParams definition being wrong? It does accept object's like that.

Not sure if it helped, but I just bumped this extension to 0.1.8 with an up-to-date version of svelte-language-server.

**Comment by Suges** [comment]
Appreciate the version bump!

Doesn't fix it...this must be a problem with the language grammar you copied in from either nova-vue or nova-typescript. One of those is missing some JavaScript rules probably.

**Comment by laosb** [comment]
> Appreciate the version bump!
> 
> 
> 
> Doesn't fix it...this must be a problem with the language grammar you copied in from either nova-vue or nova-typescript. One of those is missing some JavaScript rules probably.

I see. The bump should fix the side-problem I quoted in the comment.

Anyways, since I'm using Nova on a daily basis now, I may want to tackle this problem at some time.

**Comment by Suges** [comment]
You're right, definitely fixed that `URLSearchParams` issue!

I made a serious attempt to try to help you out on this project a few months ago, studying Panic's syntax definitions documents. I used to think I was smart until I read those; I can't make heads or tails of them.

**Comment by laosb** [comment]
A little heads up: Received my Nova 10 beta invitation today, would try to use the shiny new `tree-sitter`-based syntax highlighting!

**Comment by laosb** [comment]
First tryouts of `tree-sitter` syntax is promising.

**Comment by laosb** [comment]
Fixed in v0.2.0 with the new `tree-sitter`-based syntax🌲.
